subxt/src/backend/legacy/mod.rs

```diff
@@ -95,28 +95,36 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for LegacyBackend<T> {
         keys: Vec<Vec<u8>>,
         at: T::Hash,
     ) -> Result<StreamOfResults<StorageResponse>, Error> {
-        retry(|| async {
-            let keys = keys.clone();
-            let methods = self.methods.clone();
-
-            // For each key, return it + a future to get the result.
-            let iter = keys.into_iter().map(move |key| {
+        fn get_entry<T: Config>(
```

Comment:
I don't follow what you fixed with this but I think the previous code would try to fetch all storage keys in one retry block i.e, all storage key calls needs to be successful or retry it.

Looks more efficient to try each storage entry individually as you do now but maybe some other issue with it that I don't follow?

Reply:
Let's say we tried to fetch 2 entries:
- "A" and "B".
- "A" returns an OK result
- "B" returns an error
The function would return Ok(streamOf(OK,Err)

So it just would not retry

Reply:
Yup ok, so previously we wrapped code in `retry`, but that code didn't actually do any async stuff and just returned a `StreamOf`. Now, we are retrying each individual async request to fetch a storage value. Good catch!

Reply:
Cool, you could also create a separate PR for this if you want get it in ASAP :)